diff -Nuar -r --no-dereference a/arch/riscv/lib/clear_page.S b/arch/riscv/lib/clear_page.S
--- a/arch/riscv/lib/clear_page.S	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/lib/clear_page.S	2025-05-18 01:44:32.901649920 +0800
@@ -4,9 +4,9 @@
  */
 
 #include <linux/linkage.h>
+#include <linux/export.h>
 #include <asm/asm.h>
 #include <asm/alternative-macros.h>
-#include <asm-generic/export.h>
 #include <asm/hwcap.h>
 #include <asm/insn-def.h>
 #include <asm/page.h>
