diff -Nuar -r --no-dereference a/arch/riscv/include/asm/asm.h b/arch/riscv/include/asm/asm.h
--- a/arch/riscv/include/asm/asm.h	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/include/asm/asm.h	2025-05-18 01:44:32.823649823 +0800
@@ -104,6 +104,25 @@
 .endm
 #endif /* CONFIG_SMP */
 
+.macro load_per_cpu dst ptr tmp
+	asm_per_cpu \dst \ptr \tmp
+	REG_L \dst, 0(\dst)
+.endm
+
+#ifdef CONFIG_SHADOW_CALL_STACK
+/* gp is used as the shadow call stack pointer instead */
+.macro load_global_pointer
+.endm
+#else
+/* load __global_pointer to gp */
+.macro load_global_pointer
+.option push
+.option norelax
+	la gp, __global_pointer$
+.option pop
+.endm
+#endif /* CONFIG_SHADOW_CALL_STACK */
+
 	/* save all GPs except x1 ~ x5 */
 	.macro save_from_x6_to_x31
 	REG_S x6,  PT_T1(sp)
