diff -Nuar -r --no-dereference a/arch/riscv/kvm/vcpu_fp.c b/arch/riscv/kvm/vcpu_fp.c
--- a/arch/riscv/kvm/vcpu_fp.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kvm/vcpu_fp.c	2025-05-18 01:44:32.897649915 +0800
@@ -11,7 +11,7 @@
 #include <linux/err.h>
 #include <linux/kvm_host.h>
 #include <linux/uaccess.h>
-#include <asm/hwcap.h>
+#include <asm/cpufeature.h>
 
 #ifdef CONFIG_FPU
 void kvm_riscv_vcpu_fp_reset(struct kvm_vcpu *vcpu)
