diff -Nuar -r --no-dereference a/arch/riscv/lib/memcpy.S b/arch/riscv/lib/memcpy.S
--- a/arch/riscv/lib/memcpy.S	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/lib/memcpy.S	2025-05-18 01:44:32.901649920 +0800
@@ -7,8 +7,7 @@
 #include <asm/asm.h>
 
 /* void *memcpy(void *, const void *, size_t) */
-ENTRY(__memcpy)
-WEAK(memcpy)
+SYM_FUNC_START(__memcpy)
 	move t6, a0  /* Preserve return value */
 
 	/* Defer to byte-oriented copy for small sizes */
@@ -105,6 +104,7 @@
 	bltu a1, a3, 5b
 6:
 	ret
-END(__memcpy)
+SYM_FUNC_END(__memcpy)
+SYM_FUNC_ALIAS_WEAK(memcpy, __memcpy)
 SYM_FUNC_ALIAS(__pi_memcpy, __memcpy)
 SYM_FUNC_ALIAS(__pi___memcpy, __memcpy)
