diff -Nuar -r --no-dereference a/arch/riscv/kernel/traps.c b/arch/riscv/kernel/traps.c
--- a/arch/riscv/kernel/traps.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kernel/traps.c	2025-05-18 01:44:32.889649905 +0800
@@ -151,13 +151,45 @@
 DO_ERROR_INFO(do_trap_insn_fault,
 	SIGSEGV, SEGV_ACCERR, "instruction access fault");
 
+#ifdef CONFIG_BIND_THREAD_TO_AICORES
+#include <linux/cpumask.h>
+#define AI_OPCODE_MASK0  0xFE0000FF
+#define AI_OPCODE_MATCH0 0xE200002B
+#define AI_OPCODE_MASK1  0xFE0000FF
+#define AI_OPCODE_MATCH1 0xE600002B
+#endif
+
 asmlinkage __visible __trap_section void do_trap_insn_illegal(struct pt_regs *regs)
 {
 	bool handled;
+#ifdef CONFIG_BIND_THREAD_TO_AICORES
+	u32 epc;
+	struct cpumask ai_affinity;
+#endif
 
 	if (user_mode(regs)) {
 		irqentry_enter_from_user_mode(regs);
 
+#ifdef CONFIG_BIND_THREAD_TO_AICORES
+		/* check if trapped by ai instruction */
+		__get_user(epc, (u32 __user *)regs->epc);
+		if ((epc & AI_OPCODE_MASK0) == AI_OPCODE_MATCH0 ||
+			(epc & AI_OPCODE_MASK1) == AI_OPCODE_MATCH1) {
+			local_irq_enable();
+			if(!sched_getaffinity(0, &ai_affinity)) {
+				cpumask_and(&ai_affinity, &ai_affinity, &ai_cpu_mask);
+				if(cpumask_weight(&ai_affinity) &&
+					!sched_setaffinity(0, &ai_affinity)) {
+					/* try to bind ai core successed */
+					local_irq_disable();
+					irqentry_exit_to_user_mode(regs);
+					return;
+				}
+			}
+			local_irq_disable();
+		}
+#endif
+
 		local_irq_enable();
 
 		handled = riscv_v_first_use_handler(regs);
@@ -181,14 +213,6 @@
 
 DO_ERROR_INFO(do_trap_load_fault,
 	SIGSEGV, SEGV_ACCERR, "load access fault");
-#ifndef CONFIG_RISCV_M_MODE
-DO_ERROR_INFO(do_trap_load_misaligned,
-	SIGBUS, BUS_ADRALN, "Oops - load address misaligned");
-DO_ERROR_INFO(do_trap_store_misaligned,
-	SIGBUS, BUS_ADRALN, "Oops - store (or AMO) address misaligned");
-#else
-int handle_misaligned_load(struct pt_regs *regs);
-int handle_misaligned_store(struct pt_regs *regs);
 
 asmlinkage __visible __trap_section void do_trap_load_misaligned(struct pt_regs *regs)
 {
@@ -231,7 +255,6 @@
 		irqentry_nmi_exit(regs, state);
 	}
 }
-#endif
 DO_ERROR_INFO(do_trap_store_fault,
 	SIGSEGV, SEGV_ACCERR, "store (or AMO) access fault");
 DO_ERROR_INFO(do_trap_ecall_s,
@@ -359,34 +382,10 @@
 asmlinkage void noinstr do_irq(struct pt_regs *regs)
 {
 	irqentry_state_t state = irqentry_enter(regs);
-#ifdef CONFIG_IRQ_STACKS
-	if (on_thread_stack()) {
-		ulong *sp = per_cpu(irq_stack_ptr, smp_processor_id())
-					+ IRQ_STACK_SIZE/sizeof(ulong);
-		__asm__ __volatile(
-		"addi	sp, sp, -"RISCV_SZPTR  "\n"
-		REG_S"  ra, (sp)		\n"
-		"addi	sp, sp, -"RISCV_SZPTR  "\n"
-		REG_S"  s0, (sp)		\n"
-		"addi	s0, sp, 2*"RISCV_SZPTR "\n"
-		"move	sp, %[sp]		\n"
-		"move	a0, %[regs]		\n"
-		"call	handle_riscv_irq	\n"
-		"addi	sp, s0, -2*"RISCV_SZPTR"\n"
-		REG_L"  s0, (sp)		\n"
-		"addi	sp, sp, "RISCV_SZPTR   "\n"
-		REG_L"  ra, (sp)		\n"
-		"addi	sp, sp, "RISCV_SZPTR   "\n"
-		:
-		: [sp] "r" (sp), [regs] "r" (regs)
-		: "a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7",
-		  "t0", "t1", "t2", "t3", "t4", "t5", "t6",
-#ifndef CONFIG_FRAME_POINTER
-		  "s0",
-#endif
-		  "memory");
-	} else
-#endif
+
+	if (IS_ENABLED(CONFIG_IRQ_STACKS) && on_thread_stack())
+		call_on_irq_stack(regs, handle_riscv_irq);
+	else
 		handle_riscv_irq(regs);
 
 	irqentry_exit(regs, state);
