diff -Nuar -r --no-dereference a/arch/riscv/kernel/cpu.c b/arch/riscv/kernel/cpu.c
--- a/arch/riscv/kernel/cpu.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kernel/cpu.c	2025-05-18 01:44:32.874649887 +0800
@@ -304,28 +304,30 @@
 {
 	unsigned long cpu_id = (unsigned long)v - 1;
 	struct riscv_cpuinfo *ci = per_cpu_ptr(&riscv_cpuinfo, cpu_id);
-	struct device_node *node;
-	const char *compat;
+	struct device_node *node = of_get_cpu_node(cpu_id, NULL);
+	const char *compat, *model;
 
 	seq_printf(m, "processor\t: %lu\n", cpu_id);
 	seq_printf(m, "hart\t\t: %lu\n", cpuid_to_hartid_map(cpu_id));
+
+	if (!of_property_read_string(node, "model", &model))
+		seq_printf(m, "model name\t: %s\n", model);
+
 	print_isa(m);
 	print_mmu(m);
 
 	if (acpi_disabled) {
-		node = of_get_cpu_node(cpu_id, NULL);
-
 		if (!of_property_read_string(node, "compatible", &compat) &&
 		    strcmp(compat, "riscv"))
 			seq_printf(m, "uarch\t\t: %s\n", compat);
 
-		of_node_put(node);
 	}
 
 	seq_printf(m, "mvendorid\t: 0x%lx\n", ci->mvendorid);
 	seq_printf(m, "marchid\t\t: 0x%lx\n", ci->marchid);
 	seq_printf(m, "mimpid\t\t: 0x%lx\n", ci->mimpid);
 	seq_puts(m, "\n");
+	of_node_put(node);
 
 	return 0;
 }
