diff -Nuar -r --no-dereference a/arch/riscv/kvm/main.c b/arch/riscv/kvm/main.c
--- a/arch/riscv/kvm/main.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kvm/main.c	2025-05-18 01:44:32.895649913 +0800
@@ -11,7 +11,7 @@
 #include <linux/module.h>
 #include <linux/kvm_host.h>
 #include <asm/csr.h>
-#include <asm/hwcap.h>
+#include <asm/cpufeature.h>
 #include <asm/sbi.h>
 
 long kvm_arch_dev_ioctl(struct file *filp,
