diff -Nuar -r --no-dereference a/arch/riscv/include/asm/mmiowb.h b/arch/riscv/include/asm/mmiowb.h
--- a/arch/riscv/include/asm/mmiowb.h	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/include/asm/mmiowb.h	2025-05-18 01:44:32.836649839 +0800
@@ -7,7 +7,7 @@
  * "o,w" is sufficient to ensure that all writes to the device have completed
  * before the write to the spinlock is allowed to commit.
  */
-#define mmiowb()	__asm__ __volatile__ ("fence o,w" : : : "memory");
+#define mmiowb()	RISCV_FENCE(o, w)
 
 #include <linux/smp.h>
 #include <asm-generic/mmiowb.h>
