diff -Nuar -r --no-dereference a/arch/riscv/include/asm/insn-def.h b/arch/riscv/include/asm/insn-def.h
--- a/arch/riscv/include/asm/insn-def.h	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/include/asm/insn-def.h	2025-05-18 01:44:32.830649832 +0800
@@ -196,4 +196,8 @@
 	INSN_I(OPCODE_MISC_MEM, FUNC3(2), __RD(0),		\
 	       RS1(base), SIMM12(4))
 
+#define RISCV_PAUSE	".4byte 0x100000f"
+#define ZAWRS_WRS_NTO	".4byte 0x00d00073"
+#define ZAWRS_WRS_STO	".4byte 0x01d00073"
+
 #endif /* __ASM_INSN_DEF_H */
