diff -Nuar -r --no-dereference a/arch/riscv/kernel/ptrace.c b/arch/riscv/kernel/ptrace.c
--- a/arch/riscv/kernel/ptrace.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kernel/ptrace.c	2025-05-18 01:44:32.885649900 +0800
@@ -99,8 +99,11 @@
 	 * Ensure the vector registers have been saved to the memory before
 	 * copying them to membuf.
 	 */
-	if (target == current)
-		riscv_v_vstate_save(current, task_pt_regs(current));
+	if (target == current) {
+		get_cpu_vector_context();
+		riscv_v_vstate_save(&current->thread.vstate, task_pt_regs(current));
+		put_cpu_vector_context();
+	}
 
 	ptrace_vstate.vstart = vstate->vstart;
 	ptrace_vstate.vl = vstate->vl;
