diff -Nuar -r --no-dereference a/arch/riscv/kernel/module-sections.c b/arch/riscv/kernel/module-sections.c
--- a/arch/riscv/kernel/module-sections.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kernel/module-sections.c	2025-05-18 01:44:32.880649894 +0800
@@ -55,6 +55,8 @@
 	return (unsigned long)&plt[i];
 }
 
+/* it seems that the duplicate check is not necessary */
+#ifndef CONFIG_SOC_SPACEMIT
 static int is_rela_equal(const Elf_Rela *x, const Elf_Rela *y)
 {
 	return x->r_info == y->r_info && x->r_addend == y->r_addend;
@@ -69,6 +71,7 @@
 	}
 	return false;
 }
+#endif
 
 static void count_max_entries(Elf_Rela *relas, int num,
 			      unsigned int *plts, unsigned int *gots)
@@ -78,10 +81,16 @@
 	for (i = 0; i < num; i++) {
 		type = ELF_RISCV_R_TYPE(relas[i].r_info);
 		if (type == R_RISCV_CALL_PLT) {
+#ifndef CONFIG_SOC_SPACEMIT
+/* it seems that the duplicate check is not necessary */
 			if (!duplicate_rela(relas, i))
+#endif
 				(*plts)++;
 		} else if (type == R_RISCV_GOT_HI20) {
+#ifndef CONFIG_SOC_SPACEMIT
+/* it seems that the duplicate check is not necessary */
 			if (!duplicate_rela(relas, i))
+#endif
 				(*gots)++;
 		}
 	}
