diff -Nuar -r --no-dereference a/arch/riscv/kernel/smpboot.c b/arch/riscv/kernel/smpboot.c
--- a/arch/riscv/kernel/smpboot.c	2024-11-22 22:38:37.000000000 +0800
+++ b/arch/riscv/kernel/smpboot.c	2025-05-18 01:44:32.887649903 +0800
@@ -25,8 +25,9 @@
 #include <linux/of.h>
 #include <linux/sched/task_stack.h>
 #include <linux/sched/mm.h>
+
+#include <asm/cacheflush.h>
 #include <asm/cpu_ops.h>
-#include <asm/cpufeature.h>
 #include <asm/irq.h>
 #include <asm/mmu_context.h>
 #include <asm/numa.h>
@@ -245,18 +246,20 @@
 	riscv_ipi_enable();
 
 	numa_add_cpu(curr_cpuid);
-	set_cpu_online(curr_cpuid, 1);
-	check_unaligned_access(curr_cpuid);
+	set_cpu_online(curr_cpuid, true);
 
 	if (has_vector()) {
 		if (riscv_v_setup_vsize())
 			elf_hwcap &= ~COMPAT_HWCAP_ISA_V;
 	}
 
+	riscv_user_isa_enable();
+
 	/*
 	 * Remote TLB flushes are ignored while the CPU is offline, so emit
 	 * a local TLB flush right now just in case.
 	 */
+	local_flush_icache_all();
 	local_flush_tlb_all();
 	complete(&cpu_running);
 	/*
