diff -Nuar -r --no-dereference a/drivers/pci/controller/dwc/Makefile b/drivers/pci/controller/dwc/Makefile
--- a/drivers/pci/controller/dwc/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/pci/controller/dwc/Makefile	2025-05-18 01:44:45.224665225 +0800
@@ -26,6 +26,8 @@
 obj-$(CONFIG_PCIE_UNIPHIER) += pcie-uniphier.o
 obj-$(CONFIG_PCIE_UNIPHIER_EP) += pcie-uniphier-ep.o
 obj-$(CONFIG_PCIE_VISCONTI_HOST) += pcie-visconti.o
+obj-$(CONFIG_PCIE_SPACEMIT) += pcie-spacemit.o
+obj-$(CONFIG_PCI_K1X) += pcie-k1x.o
 
 # The following drivers are for devices that use the generic ACPI
 # pci_root.c driver but don't support standard ECAM config access.
