diff -Nuar -r --no-dereference a/drivers/net/ethernet/stmicro/stmmac/Makefile b/drivers/net/ethernet/stmicro/stmmac/Makefile
--- a/drivers/net/ethernet/stmicro/stmmac/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/net/ethernet/stmicro/stmmac/Makefile	2025-05-18 01:44:42.879662322 +0800
@@ -27,6 +27,8 @@
 obj-$(CONFIG_DWMAC_STM32)	+= dwmac-stm32.o
 obj-$(CONFIG_DWMAC_SUNXI)	+= dwmac-sunxi.o
 obj-$(CONFIG_DWMAC_SUN8I)	+= dwmac-sun8i.o
+obj-$(CONFIG_DWMAC_SPACEMIT)	+= dwmac-spacemit.o
+obj-$(CONFIG_DWMAC_SPACEMIT_ETHQOS)	+= dwmac-spacemit-ethqos.o
 obj-$(CONFIG_DWMAC_DWC_QOS_ETH)	+= dwmac-dwc-qos-eth.o
 obj-$(CONFIG_DWMAC_INTEL_PLAT)	+= dwmac-intel-plat.o
 obj-$(CONFIG_DWMAC_GENERIC)	+= dwmac-generic.o
