diff -Nuar -r --no-dereference a/drivers/nvmem/Makefile b/drivers/nvmem/Makefile
--- a/drivers/nvmem/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/nvmem/Makefile	2025-05-18 01:44:45.172665161 +0800
@@ -60,6 +60,8 @@
 nvmem-sc27xx-efuse-y			:= sc27xx-efuse.o
 obj-$(CONFIG_NVMEM_SNVS_LPGPR)		+= nvmem_snvs_lpgpr.o
 nvmem_snvs_lpgpr-y			:= snvs_lpgpr.o
+obj-$(CONFIG_NVMEM_SPACEMIT_EFUSE)	+= nvmem_spacemit_efuse.o
+nvmem_spacemit_efuse-y			:= spacemit-efuse.o
 obj-$(CONFIG_NVMEM_SPMI_SDAM)		+= nvmem_qcom-spmi-sdam.o
 nvmem_qcom-spmi-sdam-y			+= qcom-spmi-sdam.o
 obj-$(CONFIG_NVMEM_SPRD_EFUSE)		+= nvmem_sprd_efuse.o
