diff -Nuar -r --no-dereference a/drivers/rtc/Kconfig b/drivers/rtc/Kconfig
--- a/drivers/rtc/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/rtc/Kconfig	2025-05-18 01:44:45.996666179 +0800
@@ -718,6 +718,13 @@
 	  This driver can also be built as a module. If so, the module
 	  will be called rtc-sd3078
 
+config RTC_DRV_SPT_PMIC
+	tristate "Spacemit spm8xxx RTC"
+	depends on MFD_SPACEMIT_PMIC
+	help
+	  If you say yes here you will get support for the
+	  RTC of Spacemit spm8xxx PMIC.
+
 endif # I2C
 
 comment "SPI RTC drivers"
@@ -1432,7 +1439,7 @@
 
 config RTC_DRV_SA1100
 	tristate "SA11x0/PXA2xx/PXA910"
-	depends on ARCH_SA1100 || ARCH_PXA || ARCH_MMP
+	depends on ARCH_SA1100 || ARCH_PXA || ARCH_MMP || SOC_SPACEMIT_K1X
 	help
 	  If you say Y here you will get access to the real time clock
 	  built into your SA11x0 or PXA2xx CPU.
