diff -Nuar -r --no-dereference a/drivers/usb/dwc3/drd.c b/drivers/usb/dwc3/drd.c
--- a/drivers/usb/dwc3/drd.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/dwc3/drd.c	2025-05-18 01:44:47.459667987 +0800
@@ -452,6 +452,17 @@
 		break;
 	case USB_ROLE_DEVICE:
 		mode = DWC3_GCTL_PRTCAP_DEVICE;
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+		if (dwc->monitor_vbus && dwc->gadget_driver && dwc->gadget)
+			usb_gadget_vbus_connect(dwc->gadget);
+		break;
+#endif
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	case USB_ROLE_NONE:
+		mode = DWC3_GCTL_PRTCAP_DEVICE;
+		if (dwc->monitor_vbus && dwc->gadget_driver && dwc->gadget)
+			usb_gadget_vbus_disconnect(dwc->gadget);
+#endif
 		break;
 	default:
 		if (dwc->role_switch_default_mode == USB_DR_MODE_HOST)
@@ -477,7 +488,14 @@
 		role = USB_ROLE_HOST;
 		break;
 	case DWC3_GCTL_PRTCAP_DEVICE:
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+		if (dwc->vbus_session)
+			role = USB_ROLE_DEVICE;
+		else
+			role = USB_ROLE_NONE;
+#else
 		role = USB_ROLE_DEVICE;
+#endif
 		break;
 	case DWC3_GCTL_PRTCAP_OTG:
 		role = dwc->current_otg_role;
