diff -Nuar -r --no-dereference a/drivers/usb/host/xhci-hub.c b/drivers/usb/host/xhci-hub.c
--- a/drivers/usb/host/xhci-hub.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/host/xhci-hub.c	2025-05-18 01:44:47.571668126 +0800
@@ -12,6 +12,7 @@
 #include <linux/slab.h>
 #include <asm/unaligned.h>
 #include <linux/bitfield.h>
+#include <linux/usb/phy.h>
 
 #include "xhci.h"
 #include "xhci-trace.h"
@@ -608,6 +609,10 @@
 		 hcd->self.busnum, port->hcd_portnum + 1, portsc);
 }
 
+#if defined(CONFIG_SOC_SPACEMIT_K1X)
+extern void dwc3_spacemit_clear_disconnect(struct device *dev);
+#endif
+
 static void xhci_clear_port_change_bit(struct xhci_hcd *xhci, u16 wValue,
 		u16 wIndex, __le32 __iomem *addr, u32 port_status)
 {
@@ -625,6 +630,10 @@
 		break;
 	case USB_PORT_FEAT_C_CONNECTION:
 		status = PORT_CSC;
+#if defined(CONFIG_SOC_SPACEMIT_K1X)
+		dwc3_spacemit_clear_disconnect(xhci_to_hcd(xhci)->
+			self.controller->parent->parent);
+#endif
 		port_change_bit = "connect";
 		break;
 	case USB_PORT_FEAT_C_OVER_CURRENT:
