diff -Nuar -r --no-dereference a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
--- a/drivers/video/backlight/pwm_bl.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/video/backlight/pwm_bl.c	2025-05-18 01:44:47.744668339 +0800
@@ -693,10 +693,7 @@
 
 static const struct dev_pm_ops pwm_backlight_pm_ops = {
 #ifdef CONFIG_PM_SLEEP
-	.suspend = pwm_backlight_suspend,
-	.resume = pwm_backlight_resume,
-	.poweroff = pwm_backlight_suspend,
-	.restore = pwm_backlight_resume,
+	SET_SYSTEM_SLEEP_PM_OPS(pwm_backlight_suspend, pwm_backlight_resume)
 #endif
 };
 
