diff -Nuar -r --no-dereference a/drivers/pci/controller/dwc/Kconfig b/drivers/pci/controller/dwc/Kconfig
--- a/drivers/pci/controller/dwc/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/pci/controller/dwc/Kconfig	2025-05-18 01:44:45.224665225 +0800
@@ -415,4 +415,39 @@
 	  Say Y here if you want PCIe controller support on Toshiba Visconti SoC.
 	  This driver supports TMPV7708 SoC.
 
+config PCIE_SPACEMIT
+	bool "Spacemit PCIe controller"
+	depends on SOC_SPACEMIT || COMPILE_TEST
+	depends on PCI_MSI_IRQ_DOMAIN
+	select PCIE_DW_HOST
+	help
+	  Say Y here to enable PCIe controller support on Spacemit SoCs. The
+	  PCIe controller uses the DesignWare core plus Spacemit hardware wrappers.
+
+config PCI_K1X
+	bool
+
+config PCI_K1X_HOST
+	bool "Spacemit K1X PCIe Controller - Host Mode"
+	depends on SOC_SPACEMIT_K1X
+	depends on PCI && PCI_MSI
+	depends on OF && HAS_IOMEM
+	select PCIE_DW_HOST
+	select PCI_K1X
+	default y
+	help
+	  Enables support for the PCIe controller in the K1X SoC to work in
+          host mode.
+
+config PCI_K1X_EP
+	bool "Spacemit K1X PCIE Controller - Endpoint Mode"
+	depends on SOC_SPACEMIT_K1X
+	depends on PCI_ENDPOINT
+	depends on OF && HAS_IOMEM
+	select PCIE_DW_EP
+	select PCI_K1X
+	help
+	  Enables support for the PCIe controller in the K1X SoC to work in
+          endpoint mode.
+
 endmenu
