diff -Nuar -r --no-dereference a/drivers/power/supply/Kconfig b/drivers/power/supply/Kconfig
--- a/drivers/power/supply/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/power/supply/Kconfig	2025-05-18 01:44:45.795665931 +0800
@@ -126,6 +126,18 @@
 	  This driver can also be built as a module. If so, the module will be
 	  called cw2015_battery.
 
+config SPACEMIT_BATTERY_CW2015
+	tristate "Spacemit CW2015 Battery driver"
+	depends on I2C
+	select REGMAP_I2C
+	help
+	  Say Y here to enable support for the cellwise cw2015
+	  battery fuel gauge (used in the Pinebook Pro & others)
+	  on spacemit chip
+
+	  This driver can also be built as a module. If so, the module will be
+	  called cw2015_battery.
+
 config BATTERY_DS2760
 	tristate "DS2760 battery driver (HP iPAQ & others)"
 	depends on W1
@@ -240,6 +252,15 @@
 	help
 	  Say Y to include support for SBS compliant battery chargers.
 
+config CHARGER_SBS_VIRTUAL
+	tristate "Virtual SBS Compliant charger"
+	depends on POWER_SUPPLY
+	help
+	  Say Y to include support for a virtual SBS compliant battery charger.
+	  This driver provides a software interface for systems where charging
+	  is handled externally, but still need to report charging status
+	  through the kernel's power supply framework.
+
 config MANAGER_SBS
 	tristate "Smart Battery System Manager"
 	depends on I2C && I2C_MUX && GPIOLIB
@@ -952,4 +973,14 @@
 	  adds support for the SMB2 switch mode battery charger found
 	  in PMI8998 and related PMICs.
 
+config CHARGER_SGM415XX
+	tristate "Charger Driver for SGM415XX Charger IC"
+	depends on I2C
+	default n
+	help
+	  Say Yes to enable SGM415XX driver.
+	  It is mainly used as cap divider charger
+	  for Mediatek Pump Express 5.0.
+	  It depends on config MTK_CHARGER
+
 endif # POWER_SUPPLY
