diff -Nuar -r --no-dereference a/drivers/usb/phy/Makefile b/drivers/usb/phy/Makefile
--- a/drivers/usb/phy/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/phy/Makefile	2025-05-18 01:44:47.615668180 +0800
@@ -23,3 +23,5 @@
 obj-$(CONFIG_USB_ULPI)			+= phy-ulpi.o
 obj-$(CONFIG_USB_ULPI_VIEWPORT)		+= phy-ulpi-viewport.o
 obj-$(CONFIG_KEYSTONE_USB_PHY)		+= phy-keystone.o
+obj-$(CONFIG_K1XCI_USB2_PHY)	+= phy-k1x-ci-usb2.o
+obj-$(CONFIG_USB_K1XCI_OTG)		+= phy-k1x-ci-otg.o
\ 文件末尾没有换行符
