diff -Nuar -r --no-dereference a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
--- a/drivers/cpufreq/cpufreq.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/cpufreq/cpufreq.c	2025-05-18 01:44:35.096652656 +0800
@@ -2811,11 +2811,31 @@
 	return ret;
 }
 
+#ifndef CONFIG_SOC_SPACEMIT_K1X
 static void remove_boost_sysfs_file(void)
 {
 	if (cpufreq_boost_supported())
 		sysfs_remove_file(cpufreq_global_kobject, &boost.attr);
 }
+#else
+void remove_boost_sysfs_file(void)
+{
+	if (cpufreq_boost_supported())
+		sysfs_remove_file(cpufreq_global_kobject, &boost.attr);
+}
+
+void remove_policy_boost_sysfs_file(struct cpufreq_policy *policy)
+{
+	if (cpufreq_driver->attr[CPUFREQ_BOOST_FREQ]) {
+		sysfs_remove_file(&policy->kobj,
+				&((cpufreq_driver->attr[CPUFREQ_BOOST_FREQ])->attr));
+	}
+
+	if (cpufreq_boost_supported()) {
+		sysfs_remove_file(&policy->kobj, &local_boost.attr);
+	}
+}
+#endif
 
 int cpufreq_enable_boost_support(void)
 {
