diff -Nuar -r --no-dereference a/drivers/dma/Kconfig b/drivers/dma/Kconfig
--- a/drivers/dma/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/dma/Kconfig	2025-05-18 01:44:35.337652956 +0800
@@ -409,6 +409,12 @@
 	  Say yes here to support the Socionext Milbeaut
 	  XDMAC device.
 
+menuconfig MMP_PDMA_DRIVER
+	bool "MMP_PDMA driver"
+	help
+	  choice mmp_pdma driver
+
+if MMP_PDMA_DRIVER
 config MMP_PDMA
 	tristate "MMP PDMA support"
 	depends on ARCH_MMP || ARCH_PXA || COMPILE_TEST
@@ -416,6 +422,27 @@
 	help
 	  Support the MMP PDMA engine for PXA and MMP platform.
 
+config MMP_PDMA_SPACEMIT_K1X
+	bool "Spacemit mmp_pdma support"
+	depends on ARCH_MMP || ARCH_PXA || COMPILE_TEST || SOC_SPACEMIT_K1X
+	select DMA_ENGINE
+	help
+	  Support the MMP PDMA engine for Spacemit-k1x platform.
+endif
+
+config SPACEMIT_PDMA_SUPPORT_64BIT
+	bool "MMP PDMA support the 64-bit address"
+	default y
+	help
+	  Support 64-bit address in the MMP PDMA
+
+config ADMA_SPACEMIT_K1X
+	bool "Spacemit adma support"
+	depends on SOC_SPACEMIT_K1X && RPMSG_VIRTIO
+	select DMA_ENGINE
+	help
+	  Support the AMDA engine for Spacemit-k1x sspa.
+
 config MMP_TDMA
 	tristate "MMP Two-Channel DMA support"
 	depends on ARCH_MMP || COMPILE_TEST
@@ -432,7 +459,7 @@
 	select DMA_VIRTUAL_CHANNELS
 	help
 	  Enable support for the MOXA ART SoC DMA controller.
- 
+
 	  Say Y here if you enabled MMP ADMA, otherwise say N.
 
 config MPC512X_DMA
@@ -733,6 +760,12 @@
 	  driver provides the dmaengine required by the DisplayPort subsystem
 	  display driver.
 
+config USERSPACE_DMA
+	bool "userspace dma driver support"
+	depends on DMA_ENGINE
+	help
+	  Support dma operation in userspace
+
 # driver files
 source "drivers/dma/bestcomm/Kconfig"
 
