diff -Nuar -r --no-dereference a/drivers/cpufreq/freq_table.c b/drivers/cpufreq/freq_table.c
--- a/drivers/cpufreq/freq_table.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/cpufreq/freq_table.c	2025-05-18 01:44:35.100652661 +0800
@@ -57,7 +57,9 @@
 	 * If the driver has set its own cpuinfo.max_freq above max_freq, leave
 	 * it as is.
 	 */
+#ifndef  CONFIG_SOC_SPACEMIT_K1X
 	if (policy->cpuinfo.max_freq < max_freq)
+#endif
 		policy->max = policy->cpuinfo.max_freq = max_freq;
 
 	if (policy->min == ~0)
