diff -Nuar -r --no-dereference a/drivers/reset/Makefile b/drivers/reset/Makefile
--- a/drivers/reset/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/reset/Makefile	2025-05-18 01:44:45.979666159 +0800
@@ -41,3 +41,6 @@
 obj-$(CONFIG_RESET_UNIPHIER_GLUE) += reset-uniphier-glue.o
 obj-$(CONFIG_RESET_ZYNQ) += reset-zynq.o
 obj-$(CONFIG_ARCH_ZYNQMP) += reset-zynqmp.o
+obj-$(CONFIG_RESET_K1PRO_SPACEMIT) += reset-spacemit-k1pro.o
+obj-$(CONFIG_RESET_K1X_SPACEMIT) += reset-spacemit-k1x.o
+obj-$(CONFIG_RESET_K1_MATRIX_SPACEMIT) += reset-spacemit-k1matrix.o
