diff -Nuar -r --no-dereference a/drivers/tty/serial/Kconfig b/drivers/tty/serial/Kconfig
--- a/drivers/tty/serial/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/tty/serial/Kconfig	2025-05-18 01:44:47.331667829 +0800
@@ -408,10 +408,9 @@
 
 config SERIAL_PXA
 	bool "PXA serial port support (DEPRECATED)"
-	depends on ARCH_PXA || ARCH_MMP
+	depends on ARCH_PXA || ARCH_MMP || SOC_SPACEMIT_K1X
 	select SERIAL_CORE
 	select SERIAL_8250_PXA if SERIAL_8250=y
-	select SERIAL_PXA_NON8250 if !SERIAL_8250=y
 	help
 	  If you have a machine based on an Intel XScale PXA2xx CPU you
 	  can enable its onboard serial ports by enabling this option.
@@ -419,9 +418,26 @@
 	  Unless you have a specific need, you should use SERIAL_8250_PXA
 	  instead of this.
 
+if !SERIAL_8250
+choice
+	prompt "PXA serial driver"
+	default SERIAL_PXA_NON8250
+	help
+	  choice PXA serial driver
+
 config SERIAL_PXA_NON8250
-	bool
+	bool "PXA driver support"
+	depends on !SERIAL_8250 || COMPILE_TEST
+	help
+	  PXA serial driver
+
+config SERIAL_PXA_SPACEMIT_K1X
+	bool "Spacemit PXA driver suppor"
 	depends on !SERIAL_8250 || COMPILE_TEST
+	help
+	  Spacemit PXA serial driver
+endchoice
+endif
 
 config SERIAL_PXA_CONSOLE
 	bool "Console on PXA serial port (DEPRECATED)"
