diff -Nuar -r --no-dereference a/drivers/extcon/Kconfig b/drivers/extcon/Kconfig
--- a/drivers/extcon/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/extcon/Kconfig	2025-05-18 01:44:35.446653091 +0800
@@ -41,6 +41,13 @@
 	  I2C and enables USB data, stereo and mono audio, video, microphone
 	  and UART data to use a common connector port.
 
+config EXTCON_USB_K1XCI
+	tristate "Spacemit K1-x USB extcon support"
+	depends on GPIOLIB || COMPILE_TEST
+	help
+	  say Y here to enable spacemit k1-x usb wakeup irq based USB cable detection extcon support.
+	  Used typically if wakeup irq is used for USB ID pin detection.
+
 config EXTCON_GPIO
 	tristate "GPIO extcon support"
 	depends on GPIOLIB || COMPILE_TEST
