diff -Nuar -r --no-dereference a/drivers/usb/dwc2/params.c b/drivers/usb/dwc2/params.c
--- a/drivers/usb/dwc2/params.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/dwc2/params.c	2025-05-18 01:44:47.455667982 +0800
@@ -266,6 +266,23 @@
 	p->hird_threshold_en = false;
 }
 
+static void dwc2_set_spacemit_params(struct dwc2_hsotg *hsotg)
+{
+	struct dwc2_core_params *p = &hsotg->params;
+
+	p->otg_caps.hnp_support = false;
+	p->otg_caps.srp_support = false;
+	p->speed = DWC2_SPEED_PARAM_HIGH;
+	p->host_rx_fifo_size = 790;
+	p->host_nperio_tx_fifo_size = 384;
+	p->host_perio_tx_fifo_size = 768;
+	p->phy_type = DWC2_PHY_TYPE_PARAM_ULPI;
+	//p->phy_type = DWC2_PHY_TYPE_PARAM_UTMI;
+	//p->phy_utmi_width = 16;
+	p->ahbcfg = GAHBCFG_HBSTLEN_INCR8 <<
+		GAHBCFG_HBSTLEN_SHIFT;
+}
+
 const struct of_device_id dwc2_of_match_table[] = {
 	{ .compatible = "brcm,bcm2835-usb", .data = dwc2_set_bcm_params },
 	{ .compatible = "hisilicon,hi6220-usb", .data = dwc2_set_his_params },
@@ -305,6 +322,8 @@
 	  .data = dwc2_set_stm32mp15_hsotg_params },
 	{ .compatible = "intel,socfpga-agilex-hsotg",
 	  .data = dwc2_set_socfpga_agilex_params },
+	{ .compatible = "spacemit,k1-pro-usb",
+	  .data = dwc2_set_spacemit_params },
 	{},
 };
 MODULE_DEVICE_TABLE(of, dwc2_of_match_table);
@@ -613,7 +632,7 @@
 			valid = 1;
 		break;
 	case DWC2_PHY_TYPE_PARAM_ULPI:
-		if ((hs_phy_type == GHWCFG2_HS_PHY_TYPE_UTMI) ||
+		if ((hs_phy_type == GHWCFG2_HS_PHY_TYPE_ULPI) ||
 		    (hs_phy_type == GHWCFG2_HS_PHY_TYPE_UTMI_ULPI))
 			valid = 1;
 		break;
