diff -Nuar -r --no-dereference a/drivers/clocksource/dw_apb_timer.c b/drivers/clocksource/dw_apb_timer.c
--- a/drivers/clocksource/dw_apb_timer.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/clocksource/dw_apb_timer.c	2025-05-18 01:44:34.989652522 +0800
@@ -34,6 +34,8 @@
 #define APBTMR_CONTROL_MODE_PERIODIC	(1 << 1)
 #define APBTMR_CONTROL_INT		(1 << 2)
 
+static raw_spinlock_t dw_apb_timer_lock;
+
 static inline struct dw_apb_clock_event_device *
 ced_to_dw_apb_ced(struct clock_event_device *evt)
 {
@@ -102,8 +104,10 @@
 		return IRQ_NONE;
 	}
 
+	raw_spin_lock(&dw_ced->timer_lock);
 	if (dw_ced->eoi)
 		dw_ced->eoi(&dw_ced->timer);
+	raw_spin_unlock(&dw_ced->timer_lock);
 
 	evt->event_handler(evt);
 	return IRQ_HANDLED;
@@ -207,15 +211,20 @@
 	u32 ctrl;
 	struct dw_apb_clock_event_device *dw_ced = ced_to_dw_apb_ced(evt);
 
+	raw_spin_lock(&dw_ced->timer_lock);
+
 	/* Disable timer */
 	ctrl = apbt_readl_relaxed(&dw_ced->timer, APBTMR_N_CONTROL);
 	ctrl &= ~APBTMR_CONTROL_ENABLE;
 	apbt_writel_relaxed(&dw_ced->timer, ctrl, APBTMR_N_CONTROL);
+
 	/* write new count */
 	apbt_writel_relaxed(&dw_ced->timer, delta, APBTMR_N_LOAD_COUNT);
 	ctrl |= APBTMR_CONTROL_ENABLE;
 	apbt_writel_relaxed(&dw_ced->timer, ctrl, APBTMR_N_CONTROL);
 
+	raw_spin_unlock(&dw_ced->timer_lock);
+
 	return 0;
 }
 
@@ -248,6 +257,8 @@
 	if (!dw_ced)
 		return NULL;
 
+	raw_spin_lock_init(&dw_ced->timer_lock);
+
 	dw_ced->timer.base = base;
 	dw_ced->timer.irq = irq;
 	dw_ced->timer.freq = freq;
@@ -272,8 +283,7 @@
 	dw_ced->ced.name = name;
 
 	dw_ced->eoi = apbt_eoi;
-	err = request_irq(irq, dw_apb_clockevent_irq,
-			  IRQF_TIMER | IRQF_IRQPOLL | IRQF_NOBALANCING,
+	err = request_irq(irq, dw_apb_clockevent_irq, IRQF_ONESHOT,
 			  dw_ced->ced.name, &dw_ced->ced);
 	if (err) {
 		pr_err("failed to request timer irq\n");
