diff -Nuar -r --no-dereference a/drivers/media/rc/Makefile b/drivers/media/rc/Makefile
--- a/drivers/media/rc/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/media/rc/Makefile	2025-05-18 01:44:41.031660032 +0800
@@ -48,6 +48,7 @@
 obj-$(CONFIG_IR_SPI) += ir-spi.o
 obj-$(CONFIG_IR_STREAMZAP) += streamzap.o
 obj-$(CONFIG_IR_SUNXI) += sunxi-cir.o
+obj-$(CONFIG_IR_SPACEMIT) += ir-spacemit.o
 obj-$(CONFIG_IR_TOY) += ir_toy.o
 obj-$(CONFIG_IR_TTUSBIR) += ttusbir.o
 obj-$(CONFIG_IR_WINBOND_CIR) += winbond-cir.o
