diff -Nuar -r --no-dereference a/drivers/watchdog/Makefile b/drivers/watchdog/Makefile
--- a/drivers/watchdog/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/watchdog/Makefile	2025-05-18 01:44:47.928668566 +0800
@@ -37,6 +37,7 @@
 # ALPHA Architecture
 
 # ARM Architecture
+obj-$(CONFIG_SPACEMIT_WATCHDOG) += k1x_wdt.o
 obj-$(CONFIG_ARM_SP805_WATCHDOG) += sp805_wdt.o
 obj-$(CONFIG_ARM_SBSA_WATCHDOG) += sbsa_gwdt.o
 obj-$(CONFIG_ARMADA_37XX_WATCHDOG) += armada_37xx_wdt.o
