diff -Nuar -r --no-dereference a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
--- a/drivers/mmc/core/core.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/mmc/core/core.c	2025-05-18 01:44:41.505660619 +0800
@@ -1421,6 +1421,12 @@
 		__pm_wakeup_event(host->ws, 5000);
 
 	host->detect_change = 1;
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	if (!(host->caps2 & MMC_CAP2_NO_MMC)) {
+		mmc_rescan(&host->detect.work);
+		return;
+	}
+#endif
 	mmc_schedule_delayed_work(&host->detect, delay);
 }
 
@@ -2084,6 +2090,11 @@
 			return 0;
 	}
 
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	if (host->ops->encrypt_config)
+		host->ops->encrypt_config(host, 0);
+#endif
+
 	/* Order's important: probe SDIO, then SD, then MMC */
 	if (!(host->caps2 & MMC_CAP2_NO_SDIO))
 		if (!mmc_attach_sdio(host))
@@ -2097,6 +2108,11 @@
 		if (!mmc_attach_mmc(host))
 			return 0;
 
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	if (host->ops->encrypt_config)
+		host->ops->encrypt_config(host, 1);
+#endif
+
 out:
 	mmc_power_off(host);
 	return -EIO;
@@ -2291,6 +2307,10 @@
 	}
 
 	mmc_gpiod_request_cd_irq(host);
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	if (host->caps2 & MMC_CAP2_DISABLE_PROBE_SCAN)
+		return;
+#endif
 	_mmc_detect_change(host, 0, false);
 }
 
