diff -Nuar -r --no-dereference a/drivers/extcon/Makefile b/drivers/extcon/Makefile
--- a/drivers/extcon/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/extcon/Makefile	2025-05-18 01:44:35.446653091 +0800
@@ -22,6 +22,7 @@
 obj-$(CONFIG_EXTCON_QCOM_SPMI_MISC) += extcon-qcom-spmi-misc.o
 obj-$(CONFIG_EXTCON_RT8973A)	+= extcon-rt8973a.o
 obj-$(CONFIG_EXTCON_SM5502)	+= extcon-sm5502.o
+obj-$(CONFIG_EXTCON_USB_K1XCI)  += extcon-k1xci.o
 obj-$(CONFIG_EXTCON_USB_GPIO)	+= extcon-usb-gpio.o
 obj-$(CONFIG_EXTCON_USBC_CROS_EC) += extcon-usbc-cros-ec.o
 obj-$(CONFIG_EXTCON_USBC_TUSB320) += extcon-usbc-tusb320.o
