diff -Nuar -r --no-dereference a/drivers/usb/gadget/udc/Makefile b/drivers/usb/gadget/udc/Makefile
--- a/drivers/usb/gadget/udc/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/gadget/udc/Makefile	2025-05-18 01:44:47.503668042 +0800
@@ -43,3 +43,5 @@
 obj-$(CONFIG_USB_BDC_UDC)	+= bdc/
 obj-$(CONFIG_USB_MAX3420_UDC)	+= max3420_udc.o
 obj-$(CONFIG_USB_CDNS2_UDC)	+= cdns2/
+obj-$(CONFIG_USB_K1X_UDC)	+= k1x_udc_core.o
+
