diff -Nuar -r --no-dereference a/drivers/i2c/busses/Kconfig b/drivers/i2c/busses/Kconfig
--- a/drivers/i2c/busses/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/i2c/busses/Kconfig	2025-05-18 01:44:39.353657949 +0800
@@ -301,6 +301,14 @@
 	  This driver can also be built as a module.  If so, the module
 	  will be called i2c-sis96x.
 
+config I2C_SPACEMIT_K1X
+	tristate "Spacemit k1x I2C adapter"
+	help
+	  Say yes if you want to use I2C interface on sapcemit k1x platform.
+
+	  This driver can also be built as a module. If so, the module will be
+	  called i2c-spacemit-k1x.
+
 config I2C_VIA
 	tristate "VIA VT82C586B"
 	depends on PCI
