diff -Nuar -r --no-dereference a/drivers/usb/dwc2/platform.c b/drivers/usb/dwc2/platform.c
--- a/drivers/usb/dwc2/platform.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/dwc2/platform.c	2025-05-18 01:44:47.455667982 +0800
@@ -450,7 +450,7 @@
 	 */
 	if (!dev->dev.dma_mask)
 		dev->dev.dma_mask = &dev->dev.coherent_dma_mask;
-	retval = dma_set_coherent_mask(&dev->dev, DMA_BIT_MASK(32));
+	retval = dma_set_mask_and_coherent(&dev->dev, DMA_BIT_MASK(40));
 	if (retval) {
 		dev_err(&dev->dev, "can't set coherent DMA mask: %d\n", retval);
 		return retval;
