diff -Nuar -r --no-dereference a/drivers/gpio/Kconfig b/drivers/gpio/Kconfig
--- a/drivers/gpio/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/gpio/Kconfig	2025-05-18 01:44:35.535653202 +0800
@@ -1818,6 +1818,15 @@
 	  This enables the GPIO simulator - a configfs-based GPIO testing
 	  driver.
 
+config GPIO_K1X
+	bool "SPACEMIT-K1X GPIO support"
+	depends on SOC_SPACEMIT_K1X
+	help
+	  Say yes here to support the K1X GPIO device.
+	  The K1X GPIO device may have several banks, and each
+	  bank control at most 32 GPIO pins. The number of banks
+	  is passed by device tree or platform data.
+
 endmenu
 
 endif
