diff -Nuar -r --no-dereference a/drivers/watchdog/Kconfig b/drivers/watchdog/Kconfig
--- a/drivers/watchdog/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/watchdog/Kconfig	2025-05-18 01:44:47.928668566 +0800
@@ -2257,4 +2257,18 @@
 	  To compile this driver as a module, choose M here: the
 	  module will be called keembay_wdt.
 
+config SPACEMIT_WATCHDOG
+	tristate "Spacemit-k1x SoC Watchdog"
+	depends on SOC_SPACEMIT_K1X
+	select WATCHDOG_CORE
+	help
+	  spacemit k1x plat SoC Watchdog timer. This will reboot your system when
+	  the timeout is reached.
+
+config K1X_WDT_TEST
+	bool "Support K1X watchdog test"
+	depends on SOC_SPACEMIT_K1X && SPACEMIT_WATCHDOG
+	help
+	  This will enable K1X watchdog timer test
+
 endif # WATCHDOG
