diff -Nuar -r --no-dereference a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
--- a/drivers/mmc/host/sdhci.c	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/mmc/host/sdhci.c	2025-05-18 01:44:41.553660679 +0800
@@ -2632,7 +2632,6 @@
 		/* Set 1.8V Signal Enable in the Host Control2 register to 0 */
 		ctrl &= ~SDHCI_CTRL_VDD_180;
 		sdhci_writew(host, ctrl, SDHCI_HOST_CONTROL2);
-
 		if (!IS_ERR(mmc->supply.vqmmc)) {
 			ret = mmc_regulator_set_vqmmc(mmc, ios);
 			if (ret < 0) {
@@ -2715,6 +2714,16 @@
 	return !(present_state & SDHCI_DATA_0_LVL_MASK);
 }
 
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+static void sdhci_encrypt_config(struct mmc_host *mmc, unsigned int enc_flag)
+{
+	struct sdhci_host *host = mmc_priv(mmc);
+
+	if (host->ops->set_encrypt_feature)
+		host->ops->set_encrypt_feature(host, enc_flag);
+}
+#endif
+
 static int sdhci_prepare_hs400_tuning(struct mmc_host *mmc, struct mmc_ios *ios)
 {
 	struct sdhci_host *host = mmc_priv(mmc);
@@ -3065,6 +3074,9 @@
 	.execute_tuning			= sdhci_execute_tuning,
 	.card_event			= sdhci_card_event,
 	.card_busy	= sdhci_card_busy,
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	.encrypt_config = sdhci_encrypt_config,
+#endif
 };
 
 /*****************************************************************************\
