diff -Nuar -r --no-dereference a/drivers/thermal/Kconfig b/drivers/thermal/Kconfig
--- a/drivers/thermal/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/thermal/Kconfig	2025-05-18 01:44:47.243667721 +0800
@@ -201,6 +201,13 @@
 	  This implements the CPU cooling mechanism through
 	  idle injection. This will throttle the CPU by injecting
 	  idle cycle.
+
+config CPU_HOTPLUG_THERMAL
+	bool "CPU hotplug cooling device"
+	depends on HOTPLUG_CPU && SOC_SPACEMIT
+	help
+	  This implements the CPU cooling mechanism through
+	  cpu hotplug.
 endif
 
 config DEVFREQ_THERMAL
@@ -440,6 +447,13 @@
 	  This driver can also be built as a module. If so, the module will
 	  be called amlogic_thermal.
 
+config K1X_THERMAL
+	tristate "Spacemit K1X Thermal Support"
+	depends on OF && SOC_SPACEMIT
+	help
+	  Enable this option if you want to have support for thermal management
+	  controller present in Spacemit SoCs
+
 menu "Intel thermal drivers"
 depends on X86 || X86_INTEL_QUARK || COMPILE_TEST
 source "drivers/thermal/intel/Kconfig"
