diff -Nuar -r --no-dereference a/drivers/iio/adc/Kconfig b/drivers/iio/adc/Kconfig
--- a/drivers/iio/adc/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/iio/adc/Kconfig	2025-05-18 01:44:39.449658068 +0800
@@ -1448,4 +1448,10 @@
 	  The driver can also be built as a module. If so, the module will be called
 	  xilinx-ams.
 
+config SPACEMIT_P1_ADC
+	tristate "Spacemit P1 adc driver"
+	depends on MFD_SPACEMIT_PMIC
+	help
+	  Say yes here to support for Spacemit P1 analog-to-digital converter.
+
 endmenu
