diff -Nuar -r --no-dereference a/drivers/spi/Makefile b/drivers/spi/Makefile
--- a/drivers/spi/Makefile	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/spi/Makefile	2025-05-18 01:44:46.608666936 +0800
@@ -43,6 +43,9 @@
 obj-$(CONFIG_SPI_CS42L43)		+= spi-cs42l43.o
 obj-$(CONFIG_SPI_DAVINCI)		+= spi-davinci.o
 obj-$(CONFIG_SPI_DLN2)			+= spi-dln2.o
+obj-$(CONFIG_SPI_DESIGNWARE_EXT)	+= spi-dw-mmio-ext.o spi-dw-espi.o
+obj-$(CONFIG_SPI_K1X)			+= spi-k1x.o spi-k1x-dma.o
+obj-$(CONFIG_SPI_K1X_QSPI)		+= spi-k1x-qspi.o
 obj-$(CONFIG_SPI_DESIGNWARE)		+= spi-dw.o
 spi-dw-y				:= spi-dw-core.o
 spi-dw-$(CONFIG_SPI_DW_DMA)		+= spi-dw-dma.o
