diff -Nuar -r --no-dereference a/drivers/usb/typec/Kconfig b/drivers/usb/typec/Kconfig
--- a/drivers/usb/typec/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/usb/typec/Kconfig	2025-05-18 01:44:47.673668251 +0800
@@ -110,6 +110,17 @@
 	  If you choose to build this driver as a dynamically linked module, the
 	  module will be called wusb3801.ko.
 
+config TYPEC_HUSB239
+	tristate "Hynetek HUSB239 Type-C DRP Port controller driver"
+	depends on I2C
+	depends on USB_ROLE_SWITCH || !USB_ROLE_SWITCH
+	help
+	  Say Y or M here if your system has Hynetek HUSB239 Type-C DRP Port
+	  controller driver.
+
+	  If you choose to build this driver as a dynamically linked module, the
+	  module will be called husb239.ko.
+
 source "drivers/usb/typec/mux/Kconfig"
 
 source "drivers/usb/typec/altmodes/Kconfig"
