diff -Nuar -r --no-dereference a/drivers/pinctrl/Kconfig b/drivers/pinctrl/Kconfig
--- a/drivers/pinctrl/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/pinctrl/Kconfig	2025-05-18 01:44:45.426665475 +0800
@@ -403,6 +403,17 @@
 	help
 	  This selects the pinctrl driver for RK805.
 
+config PINCTRL_SPACEMIT_PMIC
+	tristate "Pinctrl and GPIO driver for Spacemit PMIC"
+	depends on MFD_SPACEMIT_PMIC
+	select GPIOLIB
+	select PINMUX
+	select GENERIC_PINCTRL_GROUPS
+	select GENERIC_PINMUX_FUNCTIONS
+	select GENERIC_PINCONF
+	help
+	  This selects the pinctrl driver for spacemit pmic.
+
 config PINCTRL_ROCKCHIP
 	tristate "Rockchip gpio and pinctrl driver"
 	depends on ARCH_ROCKCHIP || COMPILE_TEST
@@ -533,5 +544,6 @@
 source "drivers/pinctrl/uniphier/Kconfig"
 source "drivers/pinctrl/visconti/Kconfig"
 source "drivers/pinctrl/vt8500/Kconfig"
+source "drivers/pinctrl/spacemit/Kconfig"
 
 endif
