diff -Nuar -r --no-dereference a/drivers/net/ethernet/realtek/Kconfig b/drivers/net/ethernet/realtek/Kconfig
--- a/drivers/net/ethernet/realtek/Kconfig	2024-11-22 22:38:37.000000000 +0800
+++ b/drivers/net/ethernet/realtek/Kconfig	2025-05-18 01:44:42.765662181 +0800
@@ -113,4 +113,26 @@
 	  To compile this driver as a module, choose M here: the module
 	  will be called r8169.  This is recommended.
 
+config R8125
+	tristate "Realtek 8125 ethernet support"
+        depends on PCI
+        help
+          Say Y here if you have a Realtek Ethernet adapter belonging to
+          the following families:
+          RTL8125 2.5GBit Ethernet
+
+          To compile this driver as a module, choose M here: the module
+          will be called r8125.  This is recommended.
+
+config R8168
+        tristate "Realtek 8168 ethernet support"
+        depends on PCI
+        help
+          Say Y here if you have a Realtek Ethernet adapter belonging to
+          the following families:
+          RTL8168 Gigabit Ethernet
+
+          To compile this driver as a module, choose M here: the module
+          will be called r8168.  This is recommended.
+
 endif # NET_VENDOR_REALTEK
