diff -Nuar -r --no-dereference a/include/linux/pm_wakeirq.h b/include/linux/pm_wakeirq.h
--- a/include/linux/pm_wakeirq.h	2024-11-22 22:38:37.000000000 +0800
+++ b/include/linux/pm_wakeirq.h	2025-05-18 01:44:49.602670632 +0800
@@ -8,6 +8,9 @@
 
 extern int dev_pm_set_wake_irq(struct device *dev, int irq);
 extern int dev_pm_set_dedicated_wake_irq(struct device *dev, int irq);
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+extern int dev_pm_set_dedicated_wake_irq_spacemit(struct device *dev, int irq, int trigger_tyep);
+#endif
 extern int dev_pm_set_dedicated_wake_irq_reverse(struct device *dev, int irq);
 extern void dev_pm_clear_wake_irq(struct device *dev);
 
@@ -23,6 +26,13 @@
 	return 0;
 }
 
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+static inline int dev_pm_set_dedicated_wake_irq_spacemit(struct device *dev, int irq)
+{
+	return 0;
+}
+#endif
+
 static inline int dev_pm_set_dedicated_wake_irq_reverse(struct device *dev, int irq)
 {
 	return 0;
