diff -Nuar -r --no-dereference a/net/wireless/reg.c b/net/wireless/reg.c
--- a/net/wireless/reg.c	2024-11-22 22:38:37.000000000 +0800
+++ b/net/wireless/reg.c	2025-05-18 01:44:51.758673289 +0800
@@ -1005,7 +1005,7 @@
 	void *db;
 
 	if (!fw) {
-		pr_info("failed to load regulatory.db\n");
+		pr_debug("failed to load regulatory.db\n");
 		set_error = -ENODATA;
 	} else if (!valid_regdb(fw->data, fw->size)) {
 		pr_info("loaded regulatory.db is malformed or signature is missing/invalid\n");
