diff -Nuar -r --no-dereference a/sound/soc/codecs/es8326.h b/sound/soc/codecs/es8326.h
--- a/sound/soc/codecs/es8326.h	2024-11-22 22:38:37.000000000 +0800
+++ b/sound/soc/codecs/es8326.h	2025-05-18 01:44:52.497674199 +0800
@@ -72,6 +72,7 @@
 #define ES8326_DAC_VOL		0x50
 #define ES8326_DRC_RECOVERY	0x53
 #define ES8326_DRC_WINSIZE	0x54
+#define ES8326_DAC_CROSSTALK	0x55
 #define ES8326_HPJACK_TIMER	0x56
 #define ES8326_HPDET_TYPE	0x57
 #define ES8326_INT_SOURCE	0x58
@@ -100,7 +101,7 @@
 #define ES8326_MUTE (3 << 0)
 
 /* ES8326_CLK_CTL */
-#define ES8326_CLK_ON (0x7f << 0)
+#define ES8326_CLK_ON (0x7e << 0)
 #define ES8326_CLK_OFF (0 << 0)
 
 /* ES8326_CLK_INV */
@@ -193,4 +194,8 @@
 #define ES8326_VERSION (1 << 0)
 #define ES8326_VERSION_B (3 << 0)
 
+#if IS_ENABLED(CONFIG_SND_SOC_SPACEMIT)
+#define SPACEMIT_CONFIG_CODEC_ES8326 1
+#endif
+
 #endif
